### PR TITLE
Temperature dependence

### PR DESCRIPTION
**Pull request by maddalenacat**
Issue being addressed:

I added the functions of the Amplifiers temperature dependence in analog_components.py
and I modified HardwareresponseIncorporator that calls analog_components.py that now also contains the temperature as a parameter. 
The gain at a temperature T for a given amplifier is given by its gain at room temperature multiplied by a correction function. The correction function is obtained empirically for one amplifier of reference (one Surface board and one DRAB + fiber + IGLU chain) by studying its gain in a climate chamber at different temperatures in the range T = [223.15 K, 323.15 K]. In figures: _surface_CH0_temp_dep.pdf, surface_CH1_temp_dep.pdf, chain_temp_dep.pdf_ are presented the reference gain measurements at all T considered. In figures: _correction_surf_CH0.pdf, correction_surf_CH1.pdf, correction_chain.pdf_ are compared the data and the corrected gain function for each T value. 
[chain_temp_dep.pdf](https://github.com/nu-radio/NuRadioMC/files/6548975/chain_temp_dep.pdf)
[correction_chain.pdf](https://github.com/nu-radio/NuRadioMC/files/6548976/correction_chain.pdf)
[surface_CH0_temp_dep.pdf](https://github.com/nu-radio/NuRadioMC/files/6548980/surface_CH0_temp_dep.pdf)
[surface_CH1_temp_dep.pdf](https://github.com/nu-radio/NuRadioMC/files/6548981/surface_CH1_temp_dep.pdf)
[correction_surf_CH0.pdf](https://github.com/nu-radio/NuRadioMC/files/6548977/correction_surf_CH0.pdf)
[correction_surf_CH1.pdf](https://github.com/nu-radio/NuRadioMC/files/6548979/correction_surf_CH1.pdf)

### CODE REVIEW

File: NuRadioReco/detector/RNO_G/analog_components.py [hunk 1]
```diff
@@ -35,8 +45,8 @@ def load_amp_response(amp_type='rno_surface', path=os.path.dirname(os.path.realp
     amp_gain_f = interp1d(ff, amp_gain_discrete, bounds_error=False, fill_value=1)
     # all requests outside of measurement range are set to 0
 
-    def get_amp_gain(freqs):
-        amp_gain = amp_gain_f(freqs)
+    def get_amp_gain(temp, freqs):
```
Comment (christophwelling) [inline NuRadioReco/detector/RNO_G/analog_components.py:48]: Can you give the temp variable a reasonable default value? That way old code still works if the temperature is not specified.

File: NuRadioReco/modules/RNO_G/hardwareResponseIncorporator.py [hunk 2]
```diff
@@ -24,7 +24,7 @@ def __init__(self):
     def begin(self, debug=False):
         self.__debug = debug
 
-    def get_filter(self, frequencies, station_id, channel_id, det, sim_to_data=False, phase_only=False, mode=None, mingainlin=None):
+    def get_filter(self, frequencies, station_id, channel_id, det, temp, sim_to_data=False, phase_only=False, mode=None, mingainlin=None):
```
Comment (christophwelling) [inline NuRadioReco/modules/RNO_G/hardwareResponseIncorporator.py:27]: Here a default value for temp would also be good.

File: NuRadioReco/detector/RNO_G/analog_components.py (hunk 1 above)
Comment (christophwelling) [reply NuRadioReco/detector/RNO_G/analog_components.py:48]: Also having temp as the second argument would be better so we don't have to change older code

File: NuRadioReco/modules/RNO_G/hardwareResponseIncorporator.py [hunk 3]
```diff
@@ -86,7 +87,7 @@ def get_filter(self, frequencies, station_id, channel_id, det, sim_to_data=False
             return 1. / (amp_response * cable_response)
 
     @register_run()
-    def run(self, evt, station, det, sim_to_data=False, phase_only=False, mode=None, mingainlin=None):
+    def run(self, evt, station, det, temp=20, sim_to_data=False, phase_only=False, mode=None, mingainlin=None):
```
Comment (christophwelling) [inline NuRadioReco/modules/RNO_G/hardwareResponseIncorporator.py:90]: Can you add a docstring for the temp parameter here as well?

**Review by christophwelling — CHANGES_REQUESTED** [review]
Thanks for implementing this, looks pretty good. I just have a few nitpicks.

**Review by lpyras — APPROVED** [review]
Thank you for implementing this. Looks good to me.

**Review by cg-laser — APPROVED** [review]
looks also good to me. A bit more documentation would be good though. I think the relevant information is "the function loads a reference measurement made at room temperature and will correct it for the temperature. The correction function was found empirically by studying a few amplifiers in detail in a climate chamber". Or whatever you did. It would also be good if you could add a plot showing the measurement of the temperature dependence, or provide a link to a reference (e.g. RNO-G meeting talk)

**Comment by cg-laser** [comment]
I just saw now that you edited your initial post. Thanks for adding the documentation! Looks good to me. 
What I also meant that you could add a bit more documentation directly into the source code that explains what the correction function does.

**Pull request merged**